src/backend/metal/src/command.rs

```diff
@@ -785,8 +786,10 @@ impl Journal {
                     soft::Pass::Compute => self.compute_commands.len(),
                     soft::Pass::Blit => self.blit_commands.len(),
                 };
-                self.passes
-                    .push((pass.clone(), range.start + offset..range.end + offset));
+                self.passes.alloc().init((
```

Comment:
Is it possible to only have `push_no_copy` to avoid repeating `alloc().init()` and make it clear what's happening here?

Reply:
nope :) this is the tricky bit - the whole point of that change is to separate value creation from the memory allocation. So the creation *has to* happen after `alloc()` is called, and wrapping it into a helper would ruin the idea.